lib/Recur/RRuleIterator.php

```diff
@@ -276,12 +284,47 @@ public function fastForward(\DateTimeInterface $dt): void
 
     /* Functions that advance the iterator {{{ */
 
+    /**
+     * Advances currentDate by the interval.
+     * The time is set from the original startDate.
+     * If the recurrence is on a day when summer time started, then the
+     * time on that day may have jumped forward, for example, from 0230 to 0330.
+     * Using the original time means that the next recurrence will be calculated
+     * based on the original start time and the day/week/month/year interval.
+     * So the start time of the next occurrence can correctly revert to 0230.
+     */
+    protected function advanceTheDate(string $interval): void
+    {
+        $this->currentDate = $this->currentDate->modify($interval.' '.$this->startDate->format('H:i:s'));
+    }
+
     /**
      * Does the processing for advancing the iterator for hourly frequency.
      */
     protected function nextHourly(): void
     {
+        $hourOfCurrentDate = (int) $this->currentDate->format('G');
         $this->currentDate = $this->currentDate->modify('+'.$this->interval.' hours');
+        if (0 === $this->hourJump) {
+            // Remember if the clock time jumped forward on the next occurrence.
+            // That happens if the next event time is on a day when summer time starts
+            // and the event time is in the non-existent hour of the day.
+            // For example, an event that normally starts at 02:30 will
+            // have to start at 03:30 on that day.
+            // If the interval is just 1 hour, then there is no "jumping back" to do.
+            // The events that day will happen, for example, at 0130 0330 0430 0530...
+            if ($this->interval > 1) {
+                $expectedHourOfNextDate = ($hourOfCurrentDate + $this->interval) % 24;
+                $actualHourOfNextDate = (int) $this->currentDate->format('G');
+                $this->hourJump = $actualHourOfNextDate - $expectedHourOfNextDate;
+            }
+        } else {
+            // The hour "jumped" for the previous occurrence, to avoid the non-existent time.
+            // currentDate got set ahead by (usually) 1 hour on that day.
+            // Adjust it back for this next occurrence.
+            $this->currentDate = $this->currentDate->sub(new \DateInterval('PT'.$this->hourJump.'H'));
+            $this->hourJump = 0;
+        }
```

Comment:
Nitpick:
With the comments it's adding a lot of lines to this function. Perhaps we can move the new code to a private function "AdjustForTimeJumps"

Reply:
Refactored into private method `adjustForTimeJumpsOfHourlyEvent`